scripts/monitor-tinker.js

```diff
@@ -0,0 +1,67 @@
+const jsdom = require("jsdom");
+const { JSDOM } = jsdom;
+
+const fs = require('fs')
+
+const url = "https://www.asus.com/us/Single-Board-Computer/Tinker-Board/"
+const reg = "(?<image>http://dlcdnet.asus.com/pub/ASUS/mb/Linux/Tinker_Board_2GB/(?<date>[0-9]{8})-tinker-board-linaro-stretch-alip-v(?<version>[0-9]\.[0-9]).zip)"
+
+// Read the input file, and parse the variable input
+try {
+    const data = fs.readFileSync('test/run-tests.sh', 'utf8')
+          .split('\n')
+          .filter(line => line.match("TINKER_IMAGE_URL=.*"))
+    var line = data[0]
+    var m = line.match(`.*=\"${reg}`)
+    var imageName = m[0]
+    var date = m.groups.date
+    var version = m.groups.version
+} catch (err) {
+    console.error(err)
+    process.exit(1)
+}
+
+JSDOM.fromURL(url, {}).then(dom => {
+    var document = dom.window.document;
+    var section = document.getElementById("tinker-board-Download");
+    var boards = section.getElementsByTagName("b")
+    for (var i=0; i<boards.length; i++) {
+        if (boards[i].textContent == "TinkerOS-Debian") {
+            var newRef = boards[i].parentElement.href
+            var regNewMatch = newRef.match(reg)
+            var regOldMatch = imageName.match(reg)
+            let vn = parseFloat(regNewMatch.groups.version)
+            let vo = parseFloat(regOldMatch.groups.version)
```

Comment:
just since it is so clear here and also for consistency: try to choose a variable type generation and stick to it (so `var` or `let / const`)... my recommendation is to default to `const` and only if you think you will change the variable content at some point fall back to `let`... IMHO that helps to think about side-effects during writing + also allows catching some errors if you write somewhere you shouldn't write to (oh and just ~var~ 😉)...
side-effect freedom is also a great reason to use `map/ reduce` as the data flow gets more explicit when assigning the result of a reducer to a variable... (which then again helps in decomposing parts into smaller functions)